nntrainer/layers/input_layer.cpp

```diff
@@ -43,10 +43,25 @@ void InputLayer::setProperty(const std::vector<std::string> &values) {
 }
 
 void InputLayer::forwarding(RunLayerContext &context, bool training) {
+
   Tensor &hidden_ = context.getOutput(SINGLE_INOUT_IDX);
+  std::unique_ptr<nntrainer::Quantizer> quantizer;
   if (!context.getInPlace()) {
     Tensor &input_ = context.getInput(SINGLE_INOUT_IDX);
-    hidden_.copyData(input_);
+    quantizer = nullptr;
+
+    switch (hidden_.getDataType()) {
+      // This will be supported in Tensor itself. Until then, we use this.
+    case Tdatatype::UINT16: {
```

Comment:
Does this patch support UINT16 only
In #2984, it seems QINT16 is also considered for an activation type.
How about adding QINT16 support as well?
```suggestion
    case Tdatatype::QINT16: 
    case Tdatatype::UINT16: {
```

Reply:
Thanks. fixed.